crmprtd/process.py

```diff
@@ -68,9 +87,19 @@ def process(connection_string, sample_size, network, is_diagnostic=False):
     Session = sessionmaker(engine)
     sesh = Session()
 
-    observations = [
-        ob for ob in [align(sesh, row, is_diagnostic) for row in rows] if ob
-    ]
+    # If a start and end date were given then filter observations
+    # Or if just a start date was given. This won't work
+    # Because start and end date have defaults so the wont be empty
+    if (start_date and end_date) or start_date:
```

Comment:
One way that you could keep your logic conceptually simpler is to use what is essentially negative and positive infinity as your defaults. Then, barring no input from the user (or partial input, like a start date but not an end date), the comparison still works and you don't need to mess around with any conditionals/branches.

Reply:
See [`datetime.datetime.min` and `max`](https://docs.python.org/3.8/library/datetime.html#datetime.datetime.min).